docker/synckeys/synckeys.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+)
+
+const KeysFileName = "/keys/keys.json"
+const ApiKeyFile = "/sia-data/apipassword"
+
+func keys() ([]string, error) {
+	b, err := ioutil.ReadFile(KeysFileName)
+	if err != nil {
+		log.Println("Error while reading keys:", err.Error())
+		return nil, err
+	}
+	var keys []string
+	err = json.Unmarshal(b, &keys)
+	if err != nil {
+		log.Println("Error while unmarshalling keys:", err.Error())
+		return nil, err
+	}
+	log.Println("Successfully read keys.")
+	return keys, nil
+}
+
+func apiPass() (string, error) {
+	b, err := ioutil.ReadFile(ApiKeyFile)
+	if err != nil {
+		log.Println("Error while api password:", err.Error())
+		return "", err
+	}
+	log.Println("Successfully read apipassword.")
+	return strings.Trim(string(b), " \n"), nil
+}
+
+func registerKeys(siadIP string, keys []string, apipass string) error {
+	fmt.Printf("Will process %d keys.\n", len(keys))
+	client := http.Client{}
+	reqBody, _ := json.Marshal(map[string]string{})
+	for _, sk := range keys {
+		fmt.Println("Adding key:", sk)
+		reqUrl := fmt.Sprintf("http://%s:9980/skynet/addskykey?skykey=%s", siadIP, url.QueryEscape(sk))
```

Comment:
do you need the `siadIP` here? Couldn't you just call it `sia`? I thought you can ping containers using their service name.. is that not a thing?

Reply:
You are right! I didn't know about this. Here's an article: https://docs.docker.com/compose/networking/